### ISSUE TITLE
Skippable parameters interfere with mechanic used by MY

### ISSUE THREAD

**Issue by hostilefork** [metaeducation/ren-c#1007]
MY is a variant of the SHOVE operation which lets you write thing like:

    >> s: "olleh"
    >> s: my reverse
    >> s
    == "hello"

It rewires the evaluator to act something along the lines of **s: reverse s ...**, so that regardless of how many parameters the function took, it should work.  But it does so using tricks that make this first parameter appear evaluative.

That gets in the way of skippable parameters.  Here is a weird function that returns a reversed version of a string but appends a tag to it if you provide one as a first skippable parameter:

     foo: function [:t [<skip> tag!] s [text!]] [
         append reverse copy s :t
     ]

     >> foo "olleh"
     == "hello"

     >> foo <*> "olleh"
     == "hello<*>"

But if you try to use this with MY, the wiring in the evaluator isn't happy that it tried to take a quoted peek at an already evaluated parameter:

    >> s: "olleh"
    >> s: my foo
    ** Script Error: Can't quote non-literal from an evaluative source

So basically when the skippable slot is examining the argument, it notices that it's already a string that was looked up from a word.  It wanted to see that slot before the evaluation.  In other words, the trick used here needs to be a more sophisticated trick if it's going to work with skippable arguments.

This was noticed because COMPOSE uses the skippable feature, so this inhibits **my compose**.
